src/axel.c

```diff
@@ -223,11 +223,19 @@ int axel_open( axel_t *axel )
 		lseek(fd,0,SEEK_SET);
 
 		nread = read( fd, &axel->conf->num_connections, sizeof( axel->conf->num_connections ) );
-		assert( nread == sizeof( axel->conf->num_connections ) );
+		if( nread != sizeof( axel->conf->num_connections ) )
```

Comment:
I was thinking that to keep consistency with the code below, you could check if stsize is large enough **before** performing the read (and the assert). It's just a matter of taste, but that way we would use the same style as below

Reply:
another thing: should we probably delete the file and restart from scratch in this case? or better let the user deal with it?

Reply:
1. How? We don't know the size yet.
2. Let the user deal with it.

Reply:
I meant the size of the .st file. It is stored in stsize, so you can check if < sizeof( axel->conf->num_connections ) and in that case exit.

Reply:
Although we should probably also get rid of all the assert() and handle the errors better, because an error on read() is always possible...but we should not assert imho.